### PR TITLE
Update tuner to allow single value parameters

### PR DESCRIPTION
**Pull request by bchen1116**
first step to solve #1625 

We want to allow tuners to accept single values as input; however, we will not add these values to the search space. These values are then treated as fixed values. We use this to help with the implementation for #1625

### CODE REVIEW

File: evalml/tests/tuner_tests/test_skopt_tuner.py [hunk 1]
```diff
@@ -129,7 +137,6 @@ def test_skopt_tuner_invalid_parameters_score():
     tuner.add({'Mock Classifier': {'param a': 0, 'param b': 1.0, 'param c': 'option a'}}, np.inf)
     tuner.add({'Mock Classifier': {'param a': 0, 'param b': 1.0, 'param c': 'option a'}}, None)
     tuner.propose()
-    print(random_state)
```
Comment (bchen1116) [inline evalml/tests/tuner_tests/test_skopt_tuner.py:132]: remove unnecessary print statements

**Comment by chukarsten** [comment]
It seems you have a exception in grid_search_tuner that is uncovered by some of the tests you dropped.  It's only one line but you can find it in the codecov.io coverage changes.

**Review by chukarsten — APPROVED** [review]

**Review by chukarsten — CHANGES_REQUESTED** [review]
I think you might just need to address the uncovered exception and this is gucci.

**Review by jeremyliweishih — APPROVED** [review]
Looks great to me but could you add coverage making sure that the tuners can `propose` with single values as well?

**Review by chukarsten — APPROVED** [review]
Nice, thanks!

**Pull request merged**